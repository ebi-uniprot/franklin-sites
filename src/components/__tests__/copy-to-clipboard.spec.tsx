import { render, fireEvent, screen, waitFor } from '@testing-library/react';

import CopyToClipboard from '../copy-to-clipboard';

// Mock browser clipboard
Object.assign(navigator, {
  clipboard: {
    writeText: jest.fn(() => Promise.resolve()),
  },
});

describe('CopyToClipboard component', () => {
  const toCopy = 'foo bar baz';

  it('should render', () => {
    const { asFragment } = render(
      <CopyToClipboard
        textToCopy={toCopy}
        beforeCopy="Copy this"
        afterCopy="Copied it"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should change button text when after copy', async () => {
    render(
      <CopyToClipboard
        textToCopy={toCopy}
        beforeCopy="Copy this"
        afterCopy="Copied it"
      />
    );
    const beforeCopyButton = screen.getByText('Copy this');
    fireEvent.click(beforeCopyButton);
    await waitFor(() => screen.getByText('Copied it'));
  });

  it('should call onCopy when clicked', async () => {
    const onCopy = jest.fn();
    render(
      <CopyToClipboard
        textToCopy={toCopy}
        beforeCopy="Copy this"
        afterCopy="Copied it"
        onCopy={onCopy}
      />
    );
    const beforeCopyButton = screen.getByText('Copy this');
    fireEvent.click(beforeCopyButton);
    await waitFor(() => expect(onCopy).toHaveBeenCalledWith(toCopy));
  });
});
