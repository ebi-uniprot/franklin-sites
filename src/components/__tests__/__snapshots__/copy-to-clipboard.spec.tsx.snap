// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyToClipboard component should render 1`] = `
<DocumentFragment>
  <button
    class="button primary"
    type="button"
  >
    Copy this
  </button>
</DocumentFragment>
`;
